compiler_gym/envs/cgra/compile_settings.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+CGRACompileSettings = {
```

Comment:
I would suggesting using [pydantic](https://pydantic-docs.helpmanual.io/) to make this a class. It has a number of advantages: 1) docstrings can be rendered for documentation, (2) type validation and type safety, (3) automatic serialization/deserialization, (4) prevents typos

Here's an example of it in use in CompilerGym:

https://github.com/facebookresearch/CompilerGym/blob/development/compiler_gym/compiler_env_state.py#L19-L40

Reply:
BTW, love the documentation of the settings, good job 🙂